### PR TITLE
update invoice and readme

### PR DESCRIPTION
**Pull request by ankitdas13**

### CODE REVIEW

File: documents/invoice.md [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+
+**Parameters:**
+
+| Name            | Type    | Description                                                                  |
+|-----------------|---------|------------------------------------------------------------------------------|
+|type*          | string | entity type (here its invoice)                                               |
+|description        | string  | A brief description of the invoice.                      |
+|customer_id           | string  | customer id for which invoice need be raised                     |
+|customer           | object  | customer details in a object format                     |
+
```
Comment (SushanthGunjal) [inline documents/invoice.md:78]: Shouldn’t we add all the request params mentioned [here](https://razorpay.com/docs/api/payments/invoices/#request-parameters)

File: documents/invoice.md [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+      "quantity": 1,
+    }
+
+data:= map[string]interface{}{
+  "line_items": line_items,
+  "notes": map[string]interface{}{
+    "updated-key": "An updated note.",
+  },
+}
+body, err := client.Invoice.Invoice.Edit("<invoiceId>", data, nil)
```
Comment (SushanthGunjal) [inline documents/invoice.md:235]:
```suggestion
body, err := client.Invoice.Invoice.Update("<invoiceId>", data, nil)
```
nit: Better method name, which even matches the operation Update.

File: requests/request.go [hunk 3]
```diff
@@ -91,6 +91,11 @@ func (request *Request) SetTimeout(timeout int16) {
 func processResponse(response *http.Response) (map[string]interface{}, error) {
 	defer response.Body.Close()
 	body, err := ioutil.ReadAll(response.Body)
+
+	if len(body) == 0 || len(body) == 2{
+        resp :=  make(map[string]interface{})
+        return resp, nil
+    }
```
Comment (SushanthGunjal) [inline requests/request.go:98]: Why this change ?

File: resources/invoice.go [hunk 4]
```diff
@@ -29,3 +29,33 @@ func (inv *Invoice) Fetch(invoiceID string, queryParams map[string]interface{},
...
+}
+
+// Delete delete a draft invoice having the given invoiceID.
+func (s *Invoice) Delete(invoiceID string, queryParams map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
+	url := fmt.Sprintf("%s/%s", constants.INVOICE_URL, invoiceID)
+	return s.Request.Delete(url, queryParams, extraHeaders)
+}
+
+// Edit updates the invoice having the given invoiceID.
+func (s *Invoice) Edit(invoiceID string, data map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
```
Comment (SushanthGunjal) [inline resources/invoice.go:46]:
```suggestion
func (s *Invoice) Update(invoiceID string, data map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
```

File: requests/request.go (hunk 3 above)
Comment (ankitdas13) [reply requests/request.go:98]: because some of api's response is empty array and with this condition we can handle empty array

File: documents/invoice.md (hunk 2 above)
Comment (ankitdas13) [reply documents/invoice.md]: done

File: resources/invoice.go (hunk 4 above)
Comment (ankitdas13) [reply resources/invoice.go]: done even in unit test

File: requests/request.go (hunk 3 above)
Comment (SushanthGunjal) [reply requests/request.go:98]: @ankitdas13 With this **len(body) == 2** we will be sending empty empty array even when the length is 2, is this expected ?

File: resources/invoice.go (hunk 4 above)
Comment (SushanthGunjal) [reply resources/invoice.go]: Rename [this](https://github.com/razorpay/razorpay-go/pull/54/commits/71a7bb52186fff13f33226d63a4432f087b4382c#diff-d8d87577c7ce940f6a5cfb10605997f4987df4ab761796b1a7af196d0d0285b5R89) to TestInvoiceUpdate

File: requests/request.go (hunk 3 above)
Comment (ankitdas13) [reply requests/request.go:98]: @SushanthGunjal 
1 . In this [api](https://razorpay.com/docs/smart-collect/api/delete-tpv/) if you check the response it returns `null`.  for handling this i have added `len(body)==0`

2. In this [api](https://api.razorpay.com/v1/items/item_7Oy8OMV6BdEAac) we are getting empty array where `len(body)==2` count that bracket .

**Review by SushanthGunjal — APPROVED** [review]

**Comment by SushanthGunjal** [comment]
@ankitdas13 Add more details to the PR description. 
I don’t see anything in the description. 
Add something like. 
Adding Readme for Invoice, adding Update method for invoice so on and so forth.
